### PR TITLE
Change to fit for container run as nonrun

### PR DESCRIPTION
**Pull request by clyang82**
Resolve the leftover issue for nonroot user to run container. refer to https://github.com/istio/istio/issues/15984

Signed-off-by: Chun Lin Yang <clyang@cn.ibm.com>

### CODE REVIEW

**Comment by clyang82** [comment]
The three failed tests were fixed in https://github.com/istio/istio/pull/16014

File: istio-control/istio-config/templates/deployment.yaml [hunk 1]
```diff
@@ -66,6 +66,7 @@ spec:
           - --validation-webhook-config-file
           - /etc/config/validatingwebhookconfiguration.yaml
           - --monitoringPort=15014
+          - --validation-port=9443
```
Comment (costinm) [inline istio-control/istio-config/templates/deployment.yaml]: Does the flag exists in 1.2 ? Upgrade.

**Comment by costinm** [comment]
Force merge since everything is badly broken without this

File: istio-control/istio-config/templates/deployment.yaml (hunk 1 above)
Comment (clyang82) [reply istio-control/istio-config/templates/deployment.yaml]: This flag is just temporarily fix in installer. it should be not required when this PR is merged - https://github.com/istio/istio/pull/16015. You can see that there does not get the default value wile just hardcoded.
https://github.com/istio/istio/blob/8757a56331a384bae999dbdbb165eea6175ffd60/galley/cmd/galley/cmd/server.go#L160

**Pull request merged**